types/core/index.d.ts

```diff
@@ -352,7 +352,7 @@ export interface Defaults extends CoreChartOptions, ElementChartOptions {
 
 	plugins: PluginOptions;
 
-	set(scope: string, values: any): any;
+	set(scopeOrValues: string|any, values?: any): any;
```

Comment:
I think these are OK initially, but I think there might be a better way to type this such that the type is effectively
```
(values: any) => any | (scope: string, values: any) => any;
```

Reply:
You'll need to give me the complete thing, I'm not good enough in TS :)
Function overloading seems to be a thing in TS, but needs to have same number of parameters always. 

Would that work:
```ts
set(values: any, unused?: any): any;
set(scope: string, values: any): any;
```

Reply:
I'll take a look and see. Might not be possible in classes like this. I had a different thing in mind

Reply:
This should work ([playground](https://www.typescriptlang.org/play?#code/MYGwhgzhAEBiD29oG8BQ1oQKYBcAUAbmCAK5YQBc0YAdgJ4CUVtdA3OprnhMPAA5YqEHACcAljQDmAGmhFS5ZvSbV67DNnw9+gzKImToAH1V1Z8shAD8SxrZQcMGMQDNoeHHQHw32gdABeIOgAcmFxKRCGBydY6AB6eM4tXgEhfSlzYktbFRZ1OIwRXBIRGjls8gLYgF9oLBBsGMKEpM1CSspTPLVHOOKcUvKLKr6MGo4JidReGmFoF0RA6BosAHc4RDwGdlRF+AA6duRoACMwESoAFgAmaBqdvcQjrjDUrBDZE-PL6Fv7x7QIA)):

```ts
set(values: any): any;
set(scope: string, values: any): any;
```

I would really avoid usage of `any` in these type definitions and prefer `unknown`. But I think that in case of this method, we expect an object for `values` and we also return an object. So it should be something like:

```ts
set(values: Record<string, unknown>): Record<string, unknown>;
set(scope: string, values: Record<string, unknown>): Record<string, unknown>;
```

or by adding a new alias (if it doesn't already exist):

```ts
type AnyObject = Record<string, unknown>;

// ...

set(values: AnyObject): AnyObject;
set(scope: string, values: AnyObject): AnyObject;
```

Reply:
This differs only in style, right?
```ts
type AnyObject = { [key: string]: unknown };
```

It looked like we have these kinds of things.

Reply:
[Yes](https://www.typescriptlang.org/docs/handbook/advanced-types.html#mapped-types), just a matter of preference. I find `Record<K,T>` more intuitive but that may not be the case for everyone.

```ts
type Record<K extends keyof any, T> = {
  [P in K]: T;
};
```

Reply:
Record is probably more intuitive when you know typescript and the other way around when you don't.
@etimberg any preference?

Reply:
Probably, though `index.d.ts` already relies on such advanced TS types (e.g. `Partial<>`, `Pick<>`, etc.). I don't really mind which form is used since it's aliases behind `AnyObject` anyway.

Reply:
lets go with `Record`

Reply:
Record it is!